### ISSUE TITLE
/cofh replaceblocks doesn't replace air

### ISSUE THREAD

**Issue by AnrDaemon** [CoFH/Feedback#1134]
Tried to patch up ExU afterdark while I was testing some unrelated changes to the worldgen and getting some fluid interact crashes.
`/cofh replaceblocks -695 0 -897 -183 0 -385 bedrock air water flowing_water lava flowing_lava`
The result was "Nothing to replace." while I was hovering next to two holes and they clearly were in the range of the command. The water and lava that was going through them got replaced, but not the air.

**Comment by KingLemming** [comment]
Yes, air specifically is blacklisted. I can look into changing it, but I feel like that limitation had *really* good reason.

**Comment by AnrDaemon** [comment]
That limitation does more harm than good. I can think of many reasons I would need to block air, permanently or temporarily, while changing the rest of the world, but I can't think of even one reason to exclude it.

**Comment by skyboy** [comment]
Air is blacklisted because we have no way to tell the difference between a bad entry (block does not exist) and air. This is a limitation of the APIs available to us.

**Comment by AnrDaemon** [comment]
If you mean that you can't determine, if a block of air is occupied by a flower, I think that's an acceptable drawback. I don't think tools like this are meant to be used on a daily basis, or by someone with less-than-genuine intentions. Small inconvenience is a good trade for reliability of the tool.

P.S.
Also, `clearblocks` replacing blocks *with* air.

**Comment by skyboy** [comment]
If a block of air is occupied by a flower it is not a block of air, it is a block of flower.

No, what we cannot determine is if a block you have told the command to replace is a real block or an invalid entry (does not *exist*). The methods available will return the air block in that case.

**Comment by AnrDaemon** [comment]
Ahh. Yes, that makes sense AND an actually desired behavior. All too often we've had this issue of escaped inconveniences, with next to no means to deal with them.
